bigtable/metrics.go

```diff
@@ -0,0 +1,487 @@
+/*
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bigtable
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/bigtable/internal"
+	"github.com/google/uuid"
+	gax "github.com/googleapis/gax-go/v2"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"google.golang.org/api/option"
+	btpb "google.golang.org/genproto/googleapis/bigtable/v2"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	builtInMetricsMeterName = "bigtable.googleapis.com/internal/client/"
+
+	metricsPrefix         = "bigtable/"
+	locationMDKey         = "x-goog-ext-425905942-bin"
+	serverTimingMDKey     = "server-timing"
+	serverTimingValPrefix = "gfet4t7; dur="
+
+	// Monitored resource labels
+	monitoredResLabelKeyProject  = "project_id"
+	monitoredResLabelKeyInstance = "instance"
+	monitoredResLabelKeyTable    = "table"
+	monitoredResLabelKeyCluster  = "cluster"
+	monitoredResLabelKeyZone     = "zone"
+
+	// Metric labels
+	metricLabelKeyAppProfile         = "app_profile"
+	metricLabelKeyMethod             = "method"
+	metricLabelKeyOperationStatus    = "status"
+	metricLabelKeyStreamingOperation = "streaming"
+	metricLabelKeyClientName         = "client_name"
+	metricLabelKeyClientUID          = "client_uid"
+
+	// Metric names
+	metricNameOperationLatencies = "operation_latencies"
+	metricNameAttemptLatencies   = "attempt_latencies"
+	metricNameServerLatencies    = "server_latencies"
+	metricNameRetryCount         = "retry_count"
+)
+
+var (
```

Comment:
I think this needs a comment as to why these are not constants

Reply:
I think the comment for this block should be:
This are effectively const, but for testing purposes they are mutable